noir-projects/aztec-nr/aztec/src/messages/encryption/aes128.nr

```diff
@@ -34,9 +34,57 @@ use crate::{
 
 use std::aes128::aes128_encrypt;
 
-fn extract_close_to_uniformly_random_256_bits_from_ecdh_shared_secret_using_poseidon2(
+/** Computes a close-to-uniformly-random 256 bits from a given ECDH shared_secret.
+ * Unsafe, because the caller must understand how the shared_secret has been derived,
+ * otherwise usage of this function this could be insecure.
+ * See big comment within the body of the function.
```

Comment:
I see what you're going for, but think we can simplify the explanation. I'd change the description to go something like:

>This function is only known to be safe if `shared_secret` is computed by combining a random ephemeral key with an address's point.

Then we can have the body of the function explain why _that_ usecase is ok (which is sort of what it is already doing - it is quite specific. As it stands I initally thought this function was more general in its usage, but it turns out it's actually quite specific to what we're doing. So I'd just say that.